### PR TITLE
libtomcrypt: add function to free rsa keypair

### PR DESCRIPTION
**Pull request by jforissier**
Cherry-pick the libtomcrypt part of commit a1d5c81f8834 ("crypto: add
function to free rsa keypair") from optee_os master.

Signed-off-by: Jerome Forissier <jerome@forissier.org>

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
Why is this needed? Don't we normally just merge master into this branch when needed?

**Comment by jforissier** [comment]
> Why is this needed? Don't we normally just merge master into this branch when needed?

We do, but merging master does not add to the history of our local changes as shown by `git log --oneline --no-merges $BASE..$BRANCH`. https://optee.readthedocs.io/en/latest/building/gits/optee_os.html#import-branches
We want this complete history in order to apply the LTC patches to a newer version.

**Comment by jenswi-linaro** [comment]
Thanks for the explanation, I had missed that.
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: core/lib/libtomcrypt/rsa.c [hunk 1]
```diff
@@ -138,6 +138,19 @@ void crypto_acipher_free_rsa_public_key(struct rsa_public_key *s)
...
+{
+	if (!s)
+		return;
+	crypto_bignum_free(s->e);
+	crypto_bignum_free(s->d);
+	crypto_bignum_free(s->n);
+	crypto_bignum_free(s->p);
+	crypto_bignum_free(s->q);
+	crypto_bignum_free(s->qp);
+	crypto_bignum_free(s->dp);
```
Comment (etienne-lms) [inline core/lib/libtomcrypt/rsa.c:151]: missing `crypto_bignum_free(s->dq);`

**Comment by jforissier** [comment]
Update:
- Subject of first commit modified (replaced "libtomcrypt:" with "core: ltc")
- Two more commits cherry-picked from master

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
@jenswi-linaro thanks. Note, I prefer not to add tags to the commits that are unmodified cherry-picks from master to avoid ambiguities (i.e., commits 2 and 3). The first one is different though.

**Pull request merged**